### PR TITLE
[utils] Added a check for empty children for `addonTextChildren` func…

### PR DESCRIPTION
**Pull request by ekrasnopolskaia**
## Description

In some components there is an extra indent in the absence of children
Too little change for a detailed description

### CODE REVIEW

**Review by phytonmk — APPROVED** [review]

**Review by freeyoungstrong — APPROVED** [review]

**Review by phytonmk — CHANGES_REQUESTED** [review]
I have rebased and actualized the branch but as I see unit tests helped us to spot that in some cases proposed changes may break user apps.

File: semcore/button/package.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
     "@semcore/utils": "^3.37",
     "@semcore/spin": "^4",
     "@semcore/neighbor-location": "^2",
-    "@semcore/flex-box": "^4"
+    "@semcore/flex-box": "^4",
+    "@semcore/icon": "^2.16"
```
Comment (lsroman) [inline semcore/button/package.json:20]: It should be in devDependencies

File: semcore/link/__tests__/index.test.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import Calendar from '@semcore/icon/Calendar/m';
 const { cleanup, render, axe } = testing;
 const { shouldSupportClassName, shouldSupportRef } = testsShared;
 import Link from '../src';
+import CheckM from '@semcore/icon/Check/m';
```
Comment (lsroman) [inline semcore/link/__tests__/index.test.tsx:8]: This should be added to devDependencies

File: semcore/date-picker/src/components/ButtonTrigger.jsx [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+import addonText from '@semcore/utils/lib/addonText';
 import Calendar from '@semcore/icon/Calendar/m';
 import createComponent, { Root } from '@semcore/core';
 
 const ButtonTriggerRoot = ({ Children }) => {
   return (
     <Root render={BaseTrigger}>
       <ButtonTrigger.Addon tag={Calendar} />
-      {addonTextChildren(Children, ButtonTrigger.Text, ButtonTrigger.Addon)}
+      {addonText(Children, ButtonTrigger.Text, ButtonTrigger.Addon)}
```
Comment (lsroman) [inline semcore/date-picker/src/components/ButtonTrigger.jsx:11]: This code is not working correctly. See what the "addonText" function accepts

**Review by lsroman — COMMENTED** [review]
I would like to discuss again why change addonTextChildren in the DatePPicker

**Pull request merged**